From 4d2ea36762a73b3dcf86bef0e100342b02667a03 Mon Sep 17 00:00:00 2001
From: Al Bailey <Al.Bailey@windriver.com>
Date: Wed, 14 Mar 2018 10:52:29 -0500
Subject: [PATCH] Adding pci_weight_multiple to nova scheduler filter

Only adding to scheduler filter and not the cell filter
Also updated the ruby spec to expect the new value
---
 manifests/scheduler/filter.pp              | 8 ++++++++
 spec/classes/nova_scheduler_filter_spec.rb | 1 +
 2 files changed, 9 insertions(+)

diff --git a/manifests/scheduler/filter.pp b/manifests/scheduler/filter.pp
index 52179a7..c27de8a 100644
--- a/manifests/scheduler/filter.pp
+++ b/manifests/scheduler/filter.pp
@@ -99,6 +99,11 @@
 #   (optional) Separator character(s) for image property namespace and name
 #   Defaults to $::os_service_default
 #
+#  WRS
+# [*pci_weight_multiplier*]
+#   (optional) Multiplier for weighing hosts based on pci
+#   Defaults to $::os_service_default
+#
 class nova::scheduler::filter (
   $scheduler_host_manager                         = 'host_manager',
   $scheduler_max_attempts                         = '3',
@@ -122,6 +127,7 @@ class nova::scheduler::filter (
   $restrict_isolated_hosts_to_isolated_images     = $::os_service_default,
   $aggregate_image_properties_isolation_namespace = $::os_service_default,
   $aggregate_image_properties_isolation_separator = $::os_service_default,
+  $pci_weight_multiplier                          = $::os_service_default,
 ) {
 
   include ::nova::deps
@@ -213,6 +219,8 @@ class nova::scheduler::filter (
       value => $aggregate_image_properties_isolation_namespace;
     'filter_scheduler/aggregate_image_properties_isolation_separator':
       value => $aggregate_image_properties_isolation_separator;
+    'filter_scheduler/pci_weight_multiplier':
+      value => $pci_weight_multiplier;
   }
 
 }
diff --git a/spec/classes/nova_scheduler_filter_spec.rb b/spec/classes/nova_scheduler_filter_spec.rb
index 1ae0b10..037b506 100644
--- a/spec/classes/nova_scheduler_filter_spec.rb
+++ b/spec/classes/nova_scheduler_filter_spec.rb
@@ -31,6 +31,7 @@ describe 'nova::scheduler::filter' do
       it { is_expected.to contain_nova_config('filter_scheduler/restrict_isolated_hosts_to_isolated_images').with_value('<SERVICE DEFAULT>') }
       it { is_expected.to contain_nova_config('filter_scheduler/aggregate_image_properties_isolation_namespace').with_value('<SERVICE DEFAULT>') }
       it { is_expected.to contain_nova_config('filter_scheduler/aggregate_image_properties_isolation_separator').with_value('<SERVICE DEFAULT>') }
+      it { is_expected.to contain_nova_config('filter_scheduler/pci_weight_multiplier').with_value('<SERVICE DEFAULT>') }
 
     end
 
-- 
1.8.3.1

